theano/gpuarray/c_code/dnn_conv_base.c

```diff
@@ -85,6 +85,23 @@ typedef std::unordered_map<std::string, AlgoRec> AlgoCache;
 
 #line 87 "dnn_conv_base.c"
 
+#if __cplusplus < 201103L
+/* Using C standard interface (<ctime>). */
+#define theano_clock_t clock_t
+#define theano_clock() clock()
+#define theano_clock_to_milliseconds(t) ( 1000.0 * (t) / CLOCKS_PER_SEC )
+#define theano_clock_average_to_milliseconds(t, n) ( (1000.0 * (t) / (n)) / CLOCKS_PER_SEC )
+#else
+/* Using C++11 standard interface (<chrono>).
+I don't know if it's really more accurate, but at least
+it provides interfaces up to nanoseconds. */
+#include <chrono>
+#define theano_clock_t std::chrono::time_point
+#define theano_clock() std::chrono::steady_clock::now()
+#define theano_clock_to_milliseconds(t) ( std::chrono::duration_cast<std::chrono::nanoseconds>(t).count() / 1000000.0 )
+#define theano_clock_average_to_milliseconds(t, n) ( theano_clock_to_milliseconds(t) / (n) )
```

Comment:
This is probably where you lose fractions of milliseconds. 
I was using this successfully to get fractional milliseconds: 
duration.count()*1000.0